src/ensembl/production/metadata/api/genome.py

```diff
@@ -129,11 +169,18 @@ def fetch_genomes(self, genome_id=None, genome_uuid=None, assembly_accession=Non
             genome_select = genome_select.filter(Genome.genome_id.in_(genome_id))
 
         elif genome_uuid is not None:
-            genome_select = genome_select.filter(Genome.genome_uuid.in_(genome_uuid) )
+            genome_select = genome_select.filter(Genome.genome_uuid.in_(genome_uuid))
 
         elif assembly_accession is not None:
             genome_select = genome_select.filter(Assembly.accession.in_(assembly_accession))
 
+        # Handle when both assembly_name and ensembl_name are provided
+        # Todo: add genebuild_id once it's in the metadata DB (EA-1090)
+        if assembly_name is not None and ensembl_name is not None:
+            genome_select = genome_select.filter(
+                db.and_(Assembly.name.in_([assembly_name]), Organism.ensembl_name.in_([ensembl_name]))
+            )
+
```

Comment:
This is against the logic of the method.
Each separate parameter adds an individual filter which are combined at the end. 
Checking two at once is excessive and unnecessary. 
Adding assembly name is a good idea, but I would put it on its own.